src/librustdoc/html/render.rs

```diff
@@ -2304,6 +2363,53 @@ fn make_item_keywords(it: &clean::Item) -> String {
     format!("{}, {}", get_basic_keywords(), it.name.as_ref().unwrap())
 }
 
+fn get_index_search_type(item: &clean::Item,
+                         parent: Option<String>) -> Option<IndexItemFunctionType> {
+    match item.inner {
+        clean::FunctionItem(ref f) => Some(get_index_fn_type(&f.decl, None)),
+        clean::MethodItem(ref m) => Some(get_index_fn_type(&m.decl, parent)),
+        clean::TyMethodItem(ref m) => Some(get_index_fn_type(&m.decl, parent)),
+        _ => None
+    }
+}
+
+fn get_index_fn_type(decl: &clean::FnDecl, parent: Option<String>) -> IndexItemFunctionType {
```

Comment:
Instead of breaking this up as a separate function I tend to prefer modifying the above `match` to something like:

```rust
let (decl, parent) = match item.inner {
    clean::FunctionItem(ref f) => (&f.decl, None),
    clean::TyMethodItem(ref m) => (&m.decl, parent),
    ...
    _ => return None
};
// ...
```